Sources/Fuzzilli/Lifting/JavaScriptLifter.swift

```diff
@@ -2268,4 +2304,89 @@ public class JavaScriptLifter: Lifter {
             fields[fields.count - 1] += body + "}"
         }
     }
+
+    // Every possible position of label
+    struct LabelPin {
+        var beginPos: Int
+        var hasLabel: Bool
+    }
+
+    enum LabelType {
+        case loopblock
+        case ifblock
+        case switchblock
+        case tryblock
+        case codeBlock
+        case withblock
+    }
+
+    /// A structure that manages label positions within a specific control flow block (e.g., loop, if, switch, etc.).
+    public struct LabelStack {
```

Comment:
I wonder if this struct should actually be part of the ScriptWriter or the JavaScriptWriter itself. It feels a bit awkward that you need to pass in the ScriptWriter or internal state of the script writer (the current position). I think this would all go away if the ScriptWriter/JavaScriptWriter itself maintained this stack. And I think it would also remove the need for the `withScriptWriter` method above. Would that work? That way, the ScriptWriter also wound't need to expose the `insert` method which probably shouldn't be used for any other purposes.